Core/include/Acts/Detector/IInternalStructureBuilder.hpp

```diff
@@ -0,0 +1,51 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Navigation/DetectorVolumeUpdators.hpp"
+#include "Acts/Navigation/SurfaceCandidatesUpdators.hpp"
+
+#include <memory>
+#include <tuple>
+#include <vector>
+
+namespace Acts {
+class Surface;
+}
+
+namespace Acts {
+namespace Experimental {
+
+class DetectorVolume;
+
+using InternalStructure =
+    std::tuple<std::vector<std::shared_ptr<Surface>>,
+               std::vector<std::shared_ptr<DetectorVolume>>,
+               SurfaceCandidatesUpdator, DetectorVolumeUpdator>;
```

Comment:
that might be complex enough for a POD or struct+constructor

Reply:
Done.